Add documentation for uploading Packages to Our (#2167)
umbraco/UmbracoDocs: Extending/Packages/Uploading-to-Our/index.md

---
versionFrom: 7.0.0
---

# Uploading to Our

This document shows you how to upload your package to Our and share it with the world.

> To get started you first need to have an account on Our, if you don't head over to the site and [register](https://our.umbraco.com/member/Signup)

Reviewer: ```suggestion
To get started you first need to have an account on Our, if you don't head over to the site and [register](https://our.umbraco.com/member/Signup).
```

:::note

> You must earn some karma to upload a package from us before you can add your first package.

Reviewer: ```suggestion
Our uses a karma system where you get positive karma for answering forum posts, etc. To limit spam you will need to earn a little bit of karma before you can upload a package. Otherwise you can reach out to the [Umbraco support team](https://umbraco.com/contact-us/).
```

:::

### Creating your package

> Before you can upload your package, you first have to create your package. Read over the [Creating a Package](../Creating-a-Package/).

Reviewer: ```suggestion
Before you can upload your package, you first have to create your package. Read over the [Creating a Package](../Creating-a-Package/) article.
```

### Uploading the package to Our

> To get started head over to your [profile page](https://our.umbraco.com/member/profile/). It will tell you on this page if you have enough karma to upload your package, if you do then click `Add Package`. You may notice if you followed the [Creating a Package](../Creating-a-Package/) document that a lot of this information has already been entered. This isn't automatically pulled from the package file and will need to be entered again.

Reviewer: May be nice with a screenshot here? 🙂 (Or maybe even a gif?)
Can add an "images" folder inside this one and reference images like so:
```
![alt text here](images/screenshotname.png)
```

Author: Good shout, just added in a screenshot.

:::

### Creating your package

Before you can upload your package, you first have to create your package. Read over the [Creating a Package](../Creating-a-Package/) article.

### Uploading the package to Our

To get started head over to your [profile page](https://our.umbraco.com/member/profile/). It will tell you on this page if you have enough karma to upload your package, if you do then click `Add Package`.

![Your packages profile page](images/PackagesPage.png)

You may notice if you followed the [Creating a Package](../Creating-a-Package/) document that a lot of this information has already been entered. This isn't automatically pulled from the package file and will need to be entered again.

The form will save each page as you go and the package won't become live until you finish the process and mark it live on the final page.

:::warning
There is currently a bug that will stop you from seeing the saved packages in the list. Check out this [PR](https://github.com/umbraco/OurUmbraco/pull/526) for progress.
:::
	
## Package Details

| Property | Value | Note |
| -------- | ----- | ---- |
| Title | My Package | The name of your package |

> | Current version | | The version of your package. This will display on the front end in the button (regardless of file name) |

Reviewer 2: Does it make sense to add a placeholder for the "version" here? 🤔

```suggestion
| Current version | x.x.x | The version of your package. This will display on the front end in the button (regardless of file name) |
```

Author: No harm in it!

> | Package Category | | The category this package should be put into. |

Reviewer 2: ```suggestion
| Package Category | E.g. Backoffice Extension | The category this package should be put into. |
```

> | Package Description | | What your package does. Be clear and to the point about what your package does. |

Reviewer 2: ```suggestion
| Package Description | This package will extend the backoffice. | What your package does. Be clear and to the point about what your package does. |
```

| License name | MIT License | As input in Umbraco |
| License URL | http://opensource.org/licenses/MIT | As input in Umbraco |

### Resources

| Property | Value | Note |
| -------- | ----- | ---- |
| Package URL | https://umbraco.com | This url will be shown as the "packages url" when others install it. Will mostly be a Github repo or similar. |
| Demonstration URL | https://umbraco.com | If you have a video, blog, audio description of how your package works in more depth, this is the place to advertise it. |
| Source code URL | https://umbraco.com | A link to the source code of the package. |
| NuGet package URL | | A link to the NuGet listing of your package if available. |
| Bug tracking URL | | If you want people to submit bugs in a specific place, pop the link in here. |

### Analytics and tracking

| Property | Value | Note |
| -------- | ----- | ---- |
| Google Analytics code | UA-11111111-1 | This will tie in stats about your package into your Google Analytics account.

### Collaboration

| Property | Value | Note |
| -------- | ----- | ---- |

> | This package is open for collaboration | | This currently sets an internal flag for the package and doesn't show any different on the front end |

Reviewer 2: ```suggestion
| This package is open for collaboration | True/false | This currently sets an internal flag for the package and doesn't show any different on the front end |
```

### Package status

| Property | Value | Note |
| -------- | ----- | ---- |
| This package is retired | _Tick_ | Tick this box if you are no longer actively maintaining the package. |
| Retired message | I no longer have time to work on this package. | A quick message to explain why you are not maintaining the package. This will be shown to your Our package listing |

## Package Files

This is where you upload your package zip along with any other files you'd like the community to see. You can have multiple versions of your package uploaded at any one time.

| Property | Value | Note |
| -------- | ----- | ---- |

> | File Upload | | Select the file from your local computer |

Reviewer 2: ```suggestion
| File Upload | *File upload* | Select the file from your local computer |
```
